automation/recipes/meteor-qpsk.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# See: https://www.reddit.com/r/RTLSDR/comments/abn29d/automatic_meteor_m2_reception_on_linux_using_rtl/ed2teuv/
+# Meteor-M 2 satellite reception
+
+BASENAME=$1
+FREQUENCY=$2
+RECEIVE_TIMEOUT=$3
+
+NON_NORMALIZED_SIGNAL_FILENAME=${BASENAME}_non_normalized_signal.wav
+QPSK_FILENAME=${BASENAME}_qpsk.qpsk
+DUMP_PREFIX_FILENAME=${BASENAME}_dump
+PRODUCT_FILENAME_WITOUT_EXT=${BASENAME}_product
+PRODUCT_BITMAP_FILENAME=${PRODUCT_FILENAME_WITOUT_EXT}.bmp
+
+SIGNAL_FILENAME=${BASENAME}_signal.wav
+PRODUCT_FILENAME=${PRODUCT_FILENAME_WITOUT_EXT}.png
+
+# Record raw IQ data
+timeout $RECEIVE_TIMEOUT \
+    rtl_fm -M raw -f $FREQUENCY -s 288k -g 48 -p 1 | \
+    sox -t raw -r 288k -c 2 -b 16 -e s - -t wav $NON_NORMALIZED_SIGNAL_FILENAME rate 96k
+
+# Normalize .wav
+sox $NON_NORMALIZED_SIGNAL_FILENAME $SIGNAL_FILENAME gain -n
+
+# Demodulate .wav to QPSK
+yes | meteor-demod -o $QPSK_FILENAME $SIGNAL_FILENAME
```

Comment:
After each step there should be a check if the output file is indeed created. Otherwise if we don't the the result file, we won't know what exactly broke down.

Reply:
Fixed.

Reply:
I checked my solution and I noticed that some tools not return exit status correctly.
For example when errors occurs in noaa-apt then it prints error on STDOUT, but return exit code 0 and print nothing in STDERR. I cannot recognize that something fail. But I cannot fix it.

Reply:
Thanks for doing that check. Ok, resolving.